components/transfer/types/index.d.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react'
+import { InputChangeHandler, InputEventHandler } from '@dhis2-ui/input'
+
+export interface TransferProps {
+    options: TransferOption[]
+    onChange: (payload: { selected: string[] }) => void
+    addAllText?: string
+    addIndividualText?: string
+    className?: string
+    dataTest?: string
+    disabled?: boolean
+    enableOrderChange?: boolean
+    filterCallback?: (
+        options: TransferOption[],
+        filter: string
+    ) => TransferOption[]
+    filterCallbackPicked?: (
+        options: TransferOption[],
+        filter: string
+    ) => TransferOption[]
+    filterLabel?: string
+    filterLabelPicked?: string
+    filterPlaceholder?: string
+    filterPlaceholderPicked?: string
+    filterable?: boolean
+    filterablePicked?: boolean
+    height?: string
+    hideFilterInput?: boolean
+    hideFilterInputPicked?: boolean
+    initialSearchTerm?: string
+    initialSearchTermPicked?: string
+    leftFooter?: React.ReactNode
+    leftHeader?: React.ReactNode
+    loading?: boolean
+    loadingPicked?: boolean
+    maxSelections?: number
+    optionsWidth?: string
+    removeAllText?: string
+    removeIndividualText?: string
+    renderOption?: (props: TransferOptionRenderProps) => JSX.Element
+    rightFooter?: React.ReactNode
+    rightHeader?: React.ReactNode
+    searchTerm?: string
+    searchTermPicked?: string
+    selected?: string[]
+    selectedEmptyComponent?: React.ReactNode
+    selectedWidth?: string
+    sourceEmptyPlaceholder?: React.ReactNode
+    onEndReached?: () => void
+    onEndReachedPicked?: () => void
+    onFilterChange?: InputChangeHandler
+    onFilterChangePicked?: InputChangeHandler
+}
+
+export const Transfer: React.FC<TransferProps>
+
+export interface TransferOptionRenderProps extends TransferOption {
+    highlighted: boolean
+    selected: boolean
+    onClick: (payload: { value: string }) => void
+    onDoubleClick: (payload: { value: string }) => void
+}
+
+export interface TransferOptionProps {
```

Comment:
`onClick` and `onDoubleClick` are missing:

```suggestion
export interface TransferOptionProps {
    onClick: ({ value }: { value: string }, event: React.MouseEvent<HTMLDivElement>) => void
    onDoubleClick: ({ value }: { value: string }, event: React.MouseEvent<HTMLDivElement>) => void
```